decisions/version-compatibility.md

```diff
@@ -131,9 +131,80 @@ So let's say 1.07 is trying to sync with 5.08, central says it's compatible with
+|            | 2 (API-3.3)     | not compatible                     | not compatible                     | not compatible                     | **compatible**                     | **compatible**                     |
 
+It seems that compatibility formula for this example is pretty self explanatory:
+
+compatible =  client API version >= backward compatible from AND client API version <= current API version.
+
+*Pros:*
+- Simpler then `Option 3` and arguable simpler then `Option 1`, this seems the simplest and the most direct way to achieve the requirement
+
+*Cons:*
+- Have to maintain another version for sync (arguable could also be a pro, simplifies sync versioning by making it more direct and decoupled form main app version)
+
+## Decision
+
+I suggest to go with `Option 4`, seems simple and direct way to meet the requirements.
 
 ## Consequences
+
+### For sync
+
+We already use `major` versioning with seperate routes, omSupply is currently using v5. We can still introduce backwards incompatible change on central server (like tweaking the shape of API), in this case instead of re-defining the route we should just increment the minor (even though technically it shouldn't' be a breaking change according to semantic versioning).
+
+Some realistic examples of `Option 3`:
+
+#### Example V5.0.2
+
+1.2.0 remote server uses V5.0.1 API.
+
+7.0.0 central server is on V5.0.1 API and is backwards compatible with V5.0.0 and up.
+
+New field is added to om_action_log, this field needs to sync with central.
+
+1.2.1 remote server now requires new API version V5.0.2
+
+In 7.0.1 this field is added and API version is incremented to V5.0.2, it is still backwards compatible with V5.0.0 (since we decided that 1.2.1 does migration on om_action_log table to add default for that field and also during sync in 1.2.1 if that field is missing a default is added)
+
+When we try to sync 1.2.1 remote server with 7.0.0 central server, central server sees that it doesn't support 1.2.1 and sync is prohibited
+
+#### Example V5.1.0
+
+1.2.1 remote server uses API version V5.0.2
+
+7.0.1 central server is on V5.0.2 API and is backwards compatible with V5.0.0 and up.
+
+The shape of API request was changed, new required field in `push` was added (this field is critical for future sync and is required now).
+
+In 1.2.2 we've added this field and now it uses V5.1.0 API
+
+7.0.2 central uses this field and it's now on V5.1.0 API with minimum backwards compatible version of V5.1.0 API
+
+### For other
+
+Option 4 of compatibility should apply to other areas:
+
+#### Native Client
+
+The client itself (electron app or android app) knows it's current version and remote server knows which native client it's compatible with and backwards compatible with, this information is encoded in broadcast attribute which client can check before connecting.
```

Comment:
Should the server know about it's clients? Probably just the wording, translating from the sync case, do you mean:

The remote server knows its current version and the versions it is compatible with. The server sends this list of versions to the client and the client decide if it can work with the remote server.

Reply:
That's what I understood Andrei's paragraph was saying. Breaking the paragraph up into separate sentences does make it clearer though

Reply:
🙏 much better: a8a1e72ea67d8690c00eb5e0b3d34bf9e2b887b8